### PR TITLE
Add node builders

### PR DESCRIPTION
**Pull request by Pyrofab**
This PR makes the configuration tree structure immutable, and provides new builders to create it.

Main changes :
- `ConfigNodeBuilder` is the main way to create nodes
- Every mutating method from `Node` has been moved to `ConfigNodeBuilder`
- `ConfigNodeBuilder` and `Node` implement a new interface, `NodeLike`

*note: this PR is built on #23 and #25*

### CODE REVIEW

File: src/main/java/me/zeroeightsix/fiber/builder/ConfigNodeBuilder.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * <p><strong>This builder should not be reused!</strong>
+ * Multiple calls to {@link #build()} will result in duplicated references.
```
[the remark this answers is not in the document]
Comment (Pyrofab) [reply src/main/java/me/zeroeightsix/fiber/builder/ConfigNodeBuilder.java]: Should we throw an exception if `build` is called twice ?

File: src/main/java/me/zeroeightsix/fiber/builder/ConfigNodeBuilder.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    public ConfigNodeBuilder name(String name) {
+        if (name == null && parent != null) throw new IllegalStateException("Cannot remove the name from a child node");
+        this.name = name;
+        return this;
+    }
+
+    public ConfigNodeBuilder comment(@Nullable String comment) {
+        this.comment = comment;
+        return this;
+    }
```
[the remark this answers is not in the document]
Comment (Pyrofab) [reply src/main/java/me/zeroeightsix/fiber/builder/ConfigNodeBuilder.java]: I believe the standard approach is `<property>()`. Although I took a look on the [Fabric API repo](https://github.com/FabricMC/fabric/search?q=Builder&type=Code), and it seems the issue is not solved there either.

[the remark this answers is not in the document]
Comment (zeroeightysix) [reply src/main/java/me/zeroeightsix/fiber/builder/ConfigNodeBuilder.java:77]: Maybe go with `with<Property>()`, to separate 'property methods' with methods like `fork`?

File: src/main/java/me/zeroeightsix/fiber/builder/ConfigNodeBuilder.java [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * <p><strong>This builder cannot be reused!</strong>
+ * Multiple calls to {@link #build()} will result in an exception being thrown.
+ */
```
Comment (zeroeightysix) [inline src/main/java/me/zeroeightsix/fiber/builder/ConfigNodeBuilder.java:18]: Now that this warning has been moved to `build`, can you remove it here?

**Comment by zeroeightysix** [comment]
Thanks!

**Pull request merged**